### PR TITLE
[Wait for nnstreamer #4080][nnstreamer][trainer] Add getting model stats information @open sesame 06/08 11:20

### PR DESCRIPTION
**Pull request by songgot**
- epoch_complete_cb is called When one epoch ends in nntrainer and
      RunStats information is retrieved from the model.
    - Send event to NNStreamer, NNStreamer waits to receive results every epoch.
    - Use getStatus and nnstreamer_trainer_notify_event()

 **Self evaluation:**
 1. Build test:   [X]Passed [ ]Failed [ ]Skipped
 2. Run test:     [X]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: nnstreamer/tensor_trainer/tensor_trainer_nntrainer.cc [hunk 1]
```diff
@@ -22,37 +22,39 @@
...
  * 2. sub-plugin receives only pre-processed tensor data.
  * 3. The current feature behavior has been tested with MNIST.
+ * 4. mnist.json has 'gst_caps' containing the information below.
+ * "gst_caps":"other/tensors, format=(string)static, framerate=(fraction)0/1,
+ *  num_tensors=(int)2, dimensions=(string)1:1:784:1.1:1:10:1,
+ *  types=(string)float32.float32"
  *
  * ## Example launch line is as below
- * gst-launch-1.0 datareposrc location=mnist_trainingSet.dat \
+ * gst-launch-1.0 datareposrc location=mnist_trainingSet.dat json=mnist.json \
```
Comment (myungjoo) [inline nnstreamer/tensor_trainer/tensor_trainer_nntrainer.cc:31]: @songgot 

Isn't the property name ```json``` strange?
```json``` is just a file format for the data repository metadata.

It'd be more meaningful if it's ```metadata=mnist.json```. You are accepting metadata in the format of json.

Comment (songgot) [reply nnstreamer/tensor_trainer/tensor_trainer_nntrainer.cc:31]: I actually thought about what to write the property name as. And I found somewhere to use json with ```--json=```.
I will consider  it. 
thank you.

**Comment by jijoongmoon** [comment]
Plz check if nnstreamer @4080 is merged.

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Comment by songgot** [comment]
Thank you very much. :) :+1: 
I will pay more attention and tell you in advance.

**Pull request merged**